samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/api/backgroundworkers/EventStoreDBSubscriptionBackgroundWorker.java

```diff
@@ -0,0 +1,71 @@
+package io.eventdriven.ecommerce.api.backgroundworkers;
+
+import com.eventstore.dbclient.EventStoreDBClient;
+import io.eventdriven.ecommerce.core.events.IEventBus;
+import io.eventdriven.ecommerce.core.subscriptions.EventStoreDBSubscriptionToAll;
+import io.eventdriven.ecommerce.core.subscriptions.ISubscriptionCheckpointRepository;
+import org.springframework.context.SmartLifecycle;
+
+public class EventStoreDBSubscriptionBackgroundWorker implements SmartLifecycle {
+  private final ISubscriptionCheckpointRepository subscriptionCheckpointRepository;
+  private final EventStoreDBClient eventStore;
+  private final IEventBus eventBus;
+  private EventStoreDBSubscriptionToAll subscription;
+
+  public EventStoreDBSubscriptionBackgroundWorker(
+    EventStoreDBClient eventStore,
+    ISubscriptionCheckpointRepository subscriptionCheckpointRepository,
+    IEventBus eventBus
+  ) {
+    this.eventStore = eventStore;
+    this.subscriptionCheckpointRepository = subscriptionCheckpointRepository;
+    this.eventBus = eventBus;
+  }
+
+  @Override
+  public void start() {
+    try {
+      subscription = new EventStoreDBSubscriptionToAll(
+        eventStore,
+        subscriptionCheckpointRepository,
+        eventBus
+      );
+      subscription.subscribeToAll();
+    } catch (Throwable e) {
+      System.out.println("Failed to start Subscription to All");
```

Comment:
zamiast println powinien być wpięty jakiś logger

Reply:
+1 you can use a logger like [Log4j2](https://logging.apache.org/log4j/2.x/).

Reply:
I added Log4J2 configuration and applied it in https://github.com/oskardudycz/EventSourcing.JVM/pull/1/commits/a23e17fff58c198517afb16624d6124d71158344.